### PR TITLE
refactor: remove unused macros

### LINKED ISSUE #13505: cleanup: unused macro definitions
While worrying about global variables, I got interested in the `State`
variable. It's only used in the `*.c` files for the most part. BUT then
I found this `CURSOR_BAR_RIGHT` macro which besides committing the sin
of sneakily depending on global state, it also seems to be unused (at
least the compilation process ran to success when I deleted it).

For some reason adding `-Wunused-macros` to the compilation flags
reports on a few other unused macros… and doesn't report on
`CURSOR_BAR_RIGHT`. I didn't check closely on whether these others
were false positives, to be sure.

So should this sneaky macro be removed (and possibly others according to
`-Wunused-macros`)? Or I missed something and it's a required macro
definition?

### PR DESCRIPTION
**Pull request by dundargoc**
Closes #13505

### CODE REVIEW

File: src/nvim/quickfix.c [hunk 1]
```diff
@@ -224,7 +224,6 @@ static char_u *e_no_more_items = (char_u *)N_("E553: No more items");
 #define IS_QF_STACK(qi)       (qi->qfl_type == QFLT_QUICKFIX)
 #define IS_LL_STACK(qi)       (qi->qfl_type == QFLT_LOCATION)
 #define IS_QF_LIST(qfl)       (qfl->qfl_type == QFLT_QUICKFIX)
-#define IS_LL_LIST(qfl)       (qfl->qfl_type == QFLT_LOCATION)
```
Comment (bfredl) [inline src/nvim/quickfix.c:227]: I'd keep this for symmetry/completeness/what if vim starts to use it (quickfix.c is one of the files that mostly follows upstream vim)

File: src/nvim/map.c [hunk 2]
```diff
@@ -25,12 +25,8 @@
 #define cstr_t_eq kh_str_hash_equal
 #define uint64_t_hash kh_int64_hash_func
 #define uint64_t_eq kh_int64_hash_equal
-#define uint32_t_hash kh_int_hash_func
```
Comment (bfredl) [inline src/nvim/map.c:28]: I will use this in a PR

File: src/nvim/eval.c [hunk 3]
```diff
@@ -256,11 +256,9 @@ static struct vimvar {
 #define vv_nr           vv_di.di_tv.vval.v_number
 #define vv_bool         vv_di.di_tv.vval.v_bool
 #define vv_special      vv_di.di_tv.vval.v_special
-#define vv_float        vv_di.di_tv.vval.v_float
```
Comment (bfredl) [inline src/nvim/eval.c:259]: hmm I suppose we have a vv for each vvar? then simplest to just keep it.

File: src/nvim/quickfix.c (hunk 1 above)
Comment (dundargoc) [reply src/nvim/quickfix.c]: fixed

**Pull request merged**